src/backend/starknet.cairo

```diff
@@ -89,6 +93,33 @@ namespace Starknet {
         let (bytecode_len, bytecode) = IAccount.bytecode(starknet_address);
         return (bytecode_len, bytecode);
     }
+
+    // @notice Populate a Environment with Starknet syscalls
+    func get_env{syscall_ptr: felt*}(
+        origin: model.Address*, gas_price: felt
+    ) -> model.Environment* {
+        alloc_locals;
+        let (block_number) = get_block_number();
+        let (block_timestamp) = get_block_timestamp();
+        let (block_hashes) = alloc();
+        // TODO: fix how blockhashes are retrieved
+        memset(block_hashes, 0, 256 * 2);
+
+        let coinbase = Uint256(
```

Comment:
should be fetched from the contract at address `block_info.sequencer_address`

Reply:
ok reading the KKT1 I see 

 ```
            let coinbase = IExternallyOwnedAccountDispatcher {
                contract_address: block_info.sequencer_address
            }
```

but I don't understand how it's possible to have such an EOA, because we start from the sequencer address (sn address), and then associate an EVM address (how?).
The address in KKT0 was hardcoded to the sequencer one though, open to fix but need more context

Reply:
ok, we didn't have such a thing in KKT0. But then, the EVM address is a constant of the program so you don't really need a syscall here, but load the constant instead?

Reply:
@Eikix @bajpai244 what did you have in mind?